### PR TITLE
Migration from Homebrew/homebrew

### PR DESCRIPTION
**Pull request by DomT4**
Don't merge until https://github.com/Homebrew/homebrew/pull/38855 is merged.

Part 1a of https://github.com/Homebrew/homebrew/issues/38819.

### CODE REVIEW

**Comment by xu-cheng** [comment]
It times out in travis. Maybe should split this PR? Also we need add test case, otherwise it will fail at `brew audit --strict`.

I will take some time to make the cache working. So we don't need to download BasicTeX every time.

**Comment by DomT4** [comment]
Yeah, I'll break it in two shortly, and throw in some tests. I use tex infrequently, but I can probably do something basic if necessary.

**Comment by DomT4** [comment]
No test possible on ` auctex ` it seems. Looks like an internal Emacs thing only, unless you can think of anything. Added a test to ` latex-mk ` and improved the ` pplatex ` one.

**Comment by xu-cheng** [comment]
I just find this. So no cache available.
> Caching Dependencies and Directories

> The features described here are currently only available for private repositories on travis-ci.com and our new container-based infrastructure.

File: Formula/latex-mk.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      \end{document}
+    EOS
+
+    (testpath/"Makefile").write <<-EOS.undent
+      NAME=test
+
+      include #{share}/latex-mk/latex.gmk
+    EOS
+
+    system "make", "pdf"
```
Comment (xu-cheng) [inline Formula/latex-mk.rb]: Just `system bin/"latexmk", "-pdf", "test.tex"`. You don't need the Makefile. :smiley:

Comment (DomT4) [inline Formula/latex-mk.rb]: It's a tool designed to make LateX projects into all its possible derivatives. The Makefile is actually part of the test of the tool here :p

Comment (xu-cheng) [inline Formula/latex-mk.rb]: I know. It's just your test case failed. (it said it cannot find `ps2pdf`). Using `latexmk -pdf`, the tex file will be compiled by pdflatex. I think it should pass the test.

Comment (DomT4) [inline Formula/latex-mk.rb]: Indeed. Not sure why - It passes completely fine consistently locally, even under sandboxing. I essentially just borrowed the instructions on testing the tool from the upstream website, heh.

Comment (xu-cheng) [inline Formula/latex-mk.rb]: `ps2pdf ` comes from ghostscript. not BasicTeX or MacTeX.

Comment (DomT4) [inline Formula/latex-mk.rb]: Argh. Of course. Life will be easier once Superenv2 lands :smile_cat:

Comment (DomT4) [inline Formula/latex-mk.rb]: > Just system bin/"latexmk", "-pdf", "test.tex"

Didn't work for me. Latex-mk threw an ` unknown command ` error. Changed it to ` system bin/"latex-mk", "test.tex" ` instead and then asserted on ` test.dvi ` existing afterwards.

Comment (xu-cheng) [inline Formula/latex-mk.rb]: OK. at the same time, you can drop the `auctex`. I have worked out a test case.

Comment (DomT4) [inline Formula/latex-mk.rb]: Go for it.

**Pull request closed without merging**